samza-core/src/main/java/org/apache/samza/execution/JobNodeConfigureGenerator.java

```diff
@@ -0,0 +1,340 @@
+    stores.forEach(sd -> configs.putAll(sd.getStorageConfigs()));
+
+    // set the execution plan in json
+    configs.put(CONFIG_INTERNAL_EXECUTION_PLAN, executionPlanJson);
+
+    // write intermediate input/output streams to configs
+    inEdges.values().stream().filter(StreamEdge::isIntermediate).forEach(edge -> configs.putAll(edge.generateConfig()));
+
+    // write serialized serde instances and stream /store serdes to configs
+    configureSerdes(configs, inEdges, outEdges, stores, jobNode);
+
+    // generate table configuration and potential side input configuration
+    configureTables(configs, config, jobNode.getTables(), inputs);
+
+    // finalize the task.inputs configuration
+    configs.put(TaskConfig.INPUT_STREAMS(), Joiner.on(',').join(inputs));
+
+    LOG.info("Job {} has generated configs {}", jobNode.getId(), configs);
+
+    // apply configure rewriters and user configure overrides
+    return applyConfigureRewritersAndOverrides(configs, config, jobNode);
+  }
+
+  private void configureBroadcastInputs(Map<String, String> configs, Config config, Set<String> broadcasts) {
+    // TODO: remove this once we support defining broadcast input stream in high-level
+    // task.broadcast.input should be generated by the planner in the future.
+    final String taskBroadcasts = config.get(TaskConfigJava.BROADCAST_INPUT_STREAMS);
+    if (StringUtils.isNoneEmpty(taskBroadcasts)) {
+      broadcasts.add(taskBroadcasts);
+    }
+    configs.put(TaskConfigJava.BROADCAST_INPUT_STREAMS, Joiner.on(',').join(broadcasts));
+  }
+
+  private void configureWindowAndJoinInterval(Map<String, String> configs, Config config,
+      Collection<OperatorSpec> reachableOperators) {
+    // set triggering interval if a window or join is defined. Only applies to high-level applications
+    if ("-1".equals(config.get(TaskConfig.WINDOW_MS(), "-1"))) {
+      long triggerInterval = computeTriggerInterval(reachableOperators);
+      LOG.info("Using triggering interval: {}", triggerInterval);
+
+      configs.put(TaskConfig.WINDOW_MS(), String.valueOf(triggerInterval));
+    }
+  }
+
+  private JobConfig applyConfigureRewritersAndOverrides(Map<String, String> configs, Config config, JobNode jobNode) {
+    String configPrefix = String.format(CONFIG_JOB_PREFIX, jobNode.getId());
+
+    // Disallow user specified job inputs/outputs. This info comes strictly from the user application.
+    Map<String, String> allowedConfigs = new HashMap<>(config);
+    if (!jobNode.isLegacyTaskApplication()) {
+      if (allowedConfigs.containsKey(TaskConfig.INPUT_STREAMS())) {
+        LOG.warn("Specifying task inputs in configuration is not allowed with Fluent API. " + "Ignoring configured value for " + TaskConfig.INPUT_STREAMS());
+        allowedConfigs.remove(TaskConfig.INPUT_STREAMS());
+      }
+    }
+
+    LOG.debug("Job {} has allowed configs {}", jobNode.getId(), allowedConfigs);
+    return new JobConfig(Util.rewriteConfig(
```

Comment:
Can this call mergeJobConfig?

Reply:
Not sure which word/variable name in the return line you are referring to? Is it referring to "extractScopedConfig"?

Reply:
Sorry, I meant can this entire line be replace by the new mergeJobConfig method? Seems to be doing a similar thing.